### ISSUE TITLE
Qxor presenting an error when compiling with make

### ISSUE THREAD

**Issue by sillasbernardo** [skilion/onedrive#533]
I followed all the steps and installed all the dependencies but when I try "sudo make", it shows the following error:

dmd -g -ofonedrive -O -L-lcurl -L-lsqlite3 -L-ldl -J. src/config.d src/itemdb.d src/log.d src/main.d src/monitor.d src/onedrive.d src/qxor.d src/selective.d src/sqlite.d src/sync.d src/upload.d src/util.d
src/qxor.d(67): Error: need `this` for `lengthInBytes` of type `immutable(ulong)`
make: *** [Makefile:32: onedrive] Error 1

**Comment by abraunegg** [comment]
@sillasbernardo 
The 'skilion' client does not receive any support (reference) and contains issues / defects that put you at risk of data loss. Despite a new version release in Oct 2020, these defects remain. If you wish to use a client which is fully supported please refer to: https://github.com/abraunegg/onedrive

Regarding your specific issue, this is a compiler issue from the 2.097.0 release 2 days ago. This is resolved in the above linked supported codebase.

The work around is to use the prior release compiler:
```
curl -fsS https://dlang.org/install.sh | bash -s dmd-2.096.1
```


**Major differences / Enhancements between codebases:**
- Existing bugs with this client are fixed
- File upload / download validation to ensure data integrity
- Support for SharePoint / Office 365 Shared Libraries
- Support for OneDrive Business Shared Folders
- Support for Docker
- Desktop notifications via libnotify
- Dry-run capability to test configuration changes
- Prevent major OneDrive accidental data deletion after configuration change
- Support National Azure Cloud instances
- Supports single & multi-tenanted applications
- Supports rate limiting of traffic

**Comment by skilion** [comment]
Fixed in 8001e1c025ddd6c2532b03305c59e407457d059b
